platform/ios/app/MBXStateManager.m

```diff
@@ -0,0 +1,44 @@
+#import "MBXStateManager.h"
+#import <Mapbox/Mapbox.h>
+#import "MBXState.h"
+#import "MBXViewController.h"
+
+@interface MBXStateManager()
+
+@property (strong, nonatomic) MBXState *currentState;
+
+@end
+
+@implementation MBXStateManager
+
++ (instancetype) sharedManager {
+    static dispatch_once_t once;
+    static MBXStateManager* sharedManager;
+    dispatch_once(&once, ^{
+        sharedManager = [[self alloc] init];
+    });
+
+    return sharedManager;
+}
+
+- (MBXState*)currentState {
+    NSData *encodedMapState = [[NSUserDefaults standardUserDefaults] objectForKey:@"mapStateKey"];
+    MBXState *currentState = (MBXState *)[NSKeyedUnarchiver unarchiveObjectWithData: encodedMapState];
+
+    _currentState = currentState;
```

Comment:
Nit: The ivar `_currentState` is unnecessary in this context because is not used elsewhere. Just return `currentState`.